### ISSUE TITLE
Missed turn in the Jungle

### ISSUE THREAD

**Issue by sigurdhes** [sandermvanvliet/RoadCaptain#62]
I was coming from the "Junge loop Switchback" planning to head up over the "Jungle loop rope bridge" instead I was routed down the "jungle loop 1". I had to take a U-turn and was back on track.

I'm running Zwift 1.24.2 from a microsoft srurface, and was using RoadCaptain 0.6.2.0.

[roadcaptain-log-2022-04-26T111709.log](https://github.com/sandermvanvliet/RoadCaptain/files/8563961/roadcaptain-log-2022-04-26T111709.log)

[Log (old 1).txt](https://github.com/sandermvanvliet/RoadCaptain/files/8563990/Log.old.1.txt)

Cannot upload Route file, so i'l paste it here:

{
  "version": "1",
  "route": {
    "name": "RoadCaptain route starting on The Mega Pretzel",
    "zwiftRouteName": "The Mega Pretzel",
    "hasCompleted": false,
    "hasStarted": false,
    "segmentSequenceIndex": 0,
    "startingSegmentId": "watopia-bambino-fondo-004-before-before",
    "nextSegmentId": null,
    "turnToNextSegment": "none",
    "currentSegmentId": null,
    "routeSegmentSequence": [
      {
        "segmentId": "watopia-bambino-fondo-004-before-before",
        "turnToNextSegment": "left",
        "nextSegmentId": "watopia-bambino-fondo-001-after-after-after-after-before-before",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-001-after-after-after-after-before-before",
        "turnToNextSegment": "right",
        "nextSegmentId": "watopia-beach-island-loop-004",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-beach-island-loop-004",
        "turnToNextSegment": "right",
        "nextSegmentId": "watopia-bambino-fondo-002-after",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-bambino-fondo-002-after",
        "turnToNextSegment": "left",
        "nextSegmentId": "watopia-bambino-fondo-001-after-after-before-after",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-bambino-fondo-001-after-after-before-after",
        "turnToNextSegment": "right",
        "nextSegmentId": "watopia-bambino-fondo-003-before-before",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-before-before",
        "turnToNextSegment": "left",
        "nextSegmentId": "watopia-ocean-lava-cliffside-loop-001",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-ocean-lava-cliffside-loop-001",
        "turnToNextSegment": "right",
        "nextSegmentId": "watopia-big-loop-rev-001-after-before-after",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-big-loop-rev-001-after-before-after",
        "turnToNextSegment": "goStraight",
        "nextSegmentId": "watopia-big-loop-rev-001-after-before-before",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-big-loop-rev-001-after-before-before",
        "turnToNextSegment": "goStraight",
        "nextSegmentId": "watopia-big-loop-rev-001-before-after",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-big-loop-rev-001-before-after",
        "turnToNextSegment": "left",
        "nextSegmentId": "watopia-bambino-fondo-003-before-after",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-before-after",
        "turnToNextSegment": "left",
        "nextSegmentId": "watopia-bambino-fondo-003-after-after-after",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-after-after-after",
        "turnToNextSegment": "goStraight",
        "nextSegmentId": "watopia-bambino-fondo-003-after-after-before",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-after-after-before",
        "turnToNextSegment": "goStraight",
        "nextSegmentId": "watopia-bambino-fondo-003-after-before-after",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-after-before-after",
        "turnToNextSegment": "right",
        "nextSegmentId": "watopia-serpentine-8-001",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-serpentine-8-001",
        "turnToNextSegment": "goStraight",
        "nextSegmentId": "watopia-bambino-fondo-003-after-after-after",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-after-after-after",
        "turnToNextSegment": "right",
        "nextSegmentId": "watopia-bambino-fondo-003-before-after",
        "direction": "atoB"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-before-after",
        "turnToNextSegment": "left",
        "nextSegmentId": "watopia-bambino-fondo-003-before-before",
        "direction": "btoA"
      },
      {
        "segmentId": "watopia-bambino-fondo-003-before-before",
        "turnToNextSegment": "none",
        "nextSegmentId": null,
        "direction": "btoA"
      }
    ],
    "world": "watopia",
    "sport": "cycling"
  }
}

**Comment by sigurdhes** [comment]
Another Issue by the end of routes. At the last segment, the trip counter in the RoadCaptain stops before the last segment, and doesn't count the distance for the last segment.

Picture is taken by the end of the last segment of the route:
![image](https://user-images.githubusercontent.com/104097267/165337313-0f7f7aa8-b08e-45de-8dd8-c0b72c93b06d.png)

**Comment by sandermvanvliet** [comment]
Yeah I noticed that yesterday when troubleshooting the missing turn at Italian villas. 
I’ll create a separate issue for that 👍

**Comment by sandermvanvliet** [comment]
Do you mean the buttons for Wave, ride on, u-turn etc?

**Comment by VNT974** [comment]
I mean the Power, Cadence, Heartrate, Speed

**Comment by sandermvanvliet** [comment]
On this issue: Zwift indicated that the turns there are Left, Right where RoadCaptain was expecting GoStraight, Right. I've ridden this segment before so I suspect there was a change here. 
The turns have been updated for this segment (at least in the direction you were going).
